sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java

```diff
@@ -329,10 +331,20 @@ public String getAccountUrl() {
     PagedFlux<BlobContainerItem> listBlobContainersWithOptionalTimeout(ListBlobContainersOptions options,
         Duration timeout) {
         throwOnAnonymousAccess();
-        Function<String, Mono<PagedResponse<BlobContainerItem>>> func =
-            marker -> listBlobContainersSegment(marker, options, timeout);
+        BiFunction<String, Integer, Mono<PagedResponse<BlobContainerItem>>> func =
+            (marker, pageSize) -> {
+                ListBlobContainersOptions finalOptions = null;
+                if (pageSize != null) {
+                    if (options == null) {
+                        finalOptions = new ListBlobContainersOptions().setMaxResultsPerPage(pageSize);
+                    } else {
+                        finalOptions = options.setMaxResultsPerPage(pageSize);
```

Comment:
does this mutate object passed by user to the api or is there a copy somewhere in between?

there's few other places with similar mutation in this PR, same question apply.

Reply:
Fixed